### PR TITLE
Fix msgspec alias defaults

### PR DESCRIPTION
**Pull request by koxudaxi**
## Summary by CodeRabbit

* **New Features**
  * Generated msgspec models now support defaults through direct, chained, nullable, collection, mapping, enum, and recursive type aliases.
  * Nested default values are constructed with appropriate conversion and independent mutable defaults.
  * GraphQL scalar aliases now support generated defaults in msgspec models.

* **Bug Fixes**
  * Unsupported, empty, and cyclic alias paths no longer trigger invalid conversion attempts.
  * Alias-based defaults now correctly create nested structured values at runtime.

### CODE REVIEW

File: tests/data/expected/main/jsonschema/msgspec_alias_default_controls_msgspec_Struct.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+class DefaultControls(Struct):
+    direct: Item | UnsetType = field(
+        default_factory=lambda: convert({'id': 1}, type=Item)
+    )
+    direct_list: list[Item] | UnsetType = field(
+        default_factory=lambda: convert([{'id': 2}], type=list[Item])
+    )
+    empty_list: list[Item] | UnsetType = field(default_factory=list)
+    numbers: Numbers | UnsetType = field(default_factory=lambda: [1, 2])
+    mapping: ItemMap | UnsetType = field(default_factory=lambda: {'item': {'id': 3}})
```
[the remark this answers is not in the document]
Comment (koxudaxi) [reply tests/data/expected/main/jsonschema/msgspec_alias_default_controls_msgspec_Struct.py:30]: This is a real pre-existing dict-of-model default limitation, rather than a mismatch introduced by this alias-object/list fix. It is already recorded in tests/main/payload_validation/conformance.py and was explicitly excluded from B31's investigated scope. The control intentionally preserves its baseline generated bytes and runtime dictionary value; changing only this golden would not change the renderer. This PR fixes direct/chained/list Struct aliases while retaining the existing mapping behavior, with real CLI/API control coverage. A mapping-default renderer change needs its own scoped fix and runtime tests, including direct references, aliases, empty mappings and recursive targets; it is not included in this PR.
